### PR TITLE
chore: Move templated code for assert_message into the stdlib

### PR DESCRIPTION
**Pull request by kevaundray**
# Description

## Problem\*

We currently have a noirc_macros crate which checks for the stdlib and inserts two functions into the stdlib. The motivation being that we did not want users to be able to see these functions. 

Introducing a new crate for this reason, does not seem to be worth it since its ~7 lines of noir code.

## Solution\*

This PR removes the noirc_macros crate and copies the noir code that was being pasted into the stdlib, directly into the stdlib

## Documentation\*

Check one:
- [ ] No documentation needed.
- [ ] Documentation included in this PR.
- [ ] **[Exceptional Case]** Documentation to be submitted in a separate PR.

# PR Checklist\*

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Review by jfecher — COMMENTED** [review]
I think this is a large improvement :+1:

File: noir_stdlib/src/lib.nr [hunk 1]
```diff
@@ -68,3 +68,11 @@ pub fn wrapping_sub<T>(x: T, y: T) -> T {
 pub fn wrapping_mul<T>(x: T, y: T) -> T {
     crate::from_field(crate::as_field(x) * crate::as_field(y))
 }
+
+#[oracle(assert_message)]
```
Comment (vezenovm) [inline noir_stdlib/src/lib.nr:72]: Can we add some comments that this is used internally by the compiler and advise against users calling this oracle directly?

Comment (TomAFrench) [reply noir_stdlib/src/lib.nr:72]: Agreed, I'm happy to merge once this is done.

**Comment by TomAFrench** [comment]
I've moved these functions into a separate file to make it clearer that users should not call anything in that file.

**Review by jfecher — APPROVED** [review]
👍

**Pull request merged**